documentation/modules/cruise-control/con-rebalance-goals-proposals.adoc

```diff
@@ -0,0 +1,180 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='con-rebalance-goals-proposals-{context}']
+= Rebalance goals and proposals
+
+To rebalance a Kafka cluster, Cruise Control uses _rebalance goals_ to generate _rebalance proposals_. 
+
+[discrete]
+== Rebalance goals
+
+Rebalance goals specify targets for resource distribution and utilization across a Kafka cluster. 
+They are organized in a customizable hierarchy, from high priority goals to low priority goals. 
+You can designate certain rebalance goals as _hard goals_ that must be satisfied--for more information, see link:#soft-hard-goals[Soft goals and hard goals].
+
+With a few exceptions, cluster rebalancing in {ProductName} supports the same goals as the Kafka Cruise Control project (where the term "optimization goals" is used). 
+Following are the supported goals, which you can reorder in priority, enable, or disable:
+
+. Rack-awareness
+. Replica capacity
+. Capacity
+.. Disk capacity
+.. Network inbound capacity
+.. Network outbound capacity
+.. CPU capacity
+. Replica distribution
+. Potential network output
+. Resource distribution
+.. Disk utilization distribution
+.. Network inbound utilization distribution
+.. Network outbound utilization distribution
+.. CPU utilization distribution
+. Leader bytes-in rate distribution
+. Topic replica distribution
+. Leader replica distribution
+. Preferred leader election
+. Intra-broker disk capacity
+. Intra-broker disk usage distribution    
+
+NOTE: "Write your own" goals, Kafka assigner goals, and Anomaly Detector are not supported in {ProductName}.
+
+For detailed information on each rebalance goal, see link:https://github.com/linkedin/cruise-control/wiki/Pluggable-Components#goals[Goals^] in the Cruise Control Wiki. 
+
+[id="soft-hard-goals"]
+[discrete]
+=== Soft goals and hard goals
+
+The following table explains the differences between soft and hard goals in cluster rebalance calculations:
+
+[cols="3*",options="header",stripes="none",separator=¦]
+|===
+
+¦Rebalance goal type
+¦Description
+¦Example
+
+¦Soft goals
+
+¦Also known as "best effort" goals, soft goals do _not_ need to be satisfied in rebalance proposals. 
+However, they are included in rebalance calculations. 
+A rebalance proposal that satisfies _all_ hard goals but violates one or more soft goals is valid.
+
+Rebalance goals are treated as soft goals, unless designated as hard goals
+
+¦Attempt to distribute a topic's replicas evenly across the cluster (topic replica distribution goal). 
+Ignore this goal if doing so will enable all hard goals to be met.
+
+¦Hard goals
+
+¦Hard goals _must_ be satisfied in rebalance proposals. 
+In other words, a rebalance proposal that does _not_ satisfy all the designated hard goals is rejected by Cruise Control and not sent to the user.
```

Comment:
even if in a rebalance proposal request you can set a flag like "skip hard goals check" and in this case, should we mention it here? wdyt @tomncooper @kyguy ?

Reply:
Yes and link to the custom goals documentation?

Reply:
Even though we don't support custom goals?

Reply:
Ah you mean the goal shipped with CC, not write your own goals [1]

[1] https://github.com/linkedin/cruise-control/wiki/Write-your-own-goals

Reply:
Rephrased as follows:

"Hard goals _must_ be satisfied in rebalance proposals unless the `skip_hard_goal_check` option is specified."

@tomncooper - I can't find any suitable information in the Cruise Control wiki.